lib/_stream_writable.js

```diff
@@ -500,6 +536,9 @@ function afterWrite(stream, state, cb) {
 
 // If there's something in the buffer waiting, then process it
 function clearBuffer(stream, state) {
+  if (state.destroyed || state.pending) {
+    return;
+  }
```

Comment:
I fear that adding this conditional might slow us down a bit. clearBuffer is incredibly hot. Can we check this elsewhere?

Reply:
I'll investigate.

Reply:
Fixed